### PR TITLE
Add with_nghttp2

### PR DESCRIPTION
**Pull request by r-darwish**
When this flag is unspecified, the build systems auto-detects whether nghttp2 is installed,
resulting a shared object which might depend on a system library that may no exist in all
machines. Since Bincrafters does not yet offer a package for nghttp2, it should be disabled in all
builds.

### CODE REVIEW

**Comment by SSE4** [comment]
I'd prefer adding an option with False default

**Comment by r-darwish** [comment]
Will do

**Comment by r-darwish** [comment]
Done

**Comment by solvingj** [comment]
Let’s add nghhttp to conan wishlist and to trello

**Comment by r-darwish** [comment]
I'm in favor, but it might take some time. Until then I suggest that we merge this pull request, as the package might be problematic because without it. I found this issue in the following scenaro:

* I tried to pack [CPR](https://github.com/whoshuu/cpr), which depends on libcurl.
* I use Arch Linux and have nghttp2 installed, because other system pacakges depends on it. 
* Arch Linux does not seperate development packages from binary packages, like Fedora and Ubuntu do. This means that I have the nghttp2 shared object as well as the include file.
* Arch Linux also have GCC 7.2, which we don't offer binaries for at the moment.
* Basically what happens when I try to build a static version of CPR, Conan compiles a static version of libcurl which detects that I have nghttp2 installed. The static library of libcurl contains calls to nghttp2 functions.  Conan doesn't tell the linker that I have to link with the static nghttp2 lib (Which I don't have anyway) so I get linker error when trying to build CPR.

**Comment by theirix** [comment]
Looks good. Linux build passes on Travis. Let's at least disable nghttp2 and then figure out problems with nghttp2 enabled. I think we could merge initial support.

**Comment by theirix** [comment]
@r-darwish , if you don't mind, I'll merge it manually to testing/7.52.1 being a base for other branches

**Comment by r-darwish** [comment]
Sure. No problem
